docs/quick-start/customize-your-own-dApp.mdx

```diff
@@ -0,0 +1,113 @@
+
+3. **Test Smart Contracts**:
+   To ensure your smart contracts work correctly, write and run tests in the `packages/snfoundry/contracts/src/test` directory. These tests can be executed using the `yarn test` command. This helps catch and fix potential errors before deployment.
+
+## Customize Frontend
+
+### Customize Nextjs App
+
+The `nextjs` directory is a critical part of your dApp, handling the frontend logic and user interface. It offers robust features and flexibility to build a dynamic and responsive application. Let's dive into the main characteristics and how you can leverage them for your project.
+
+1. **Edit Frontend Homepage**:
+   You can edit your frontend homepage in `packages/nextjs/app/page.tsx`. This file is one of the entry points of your application, and modifying it will change what users see when they visit your homepage. By customizing this file, you can tailor the first impression your users get, showcasing key features and information prominently.
+
+2. **Routing and Page Layouts**:
+   Next.js supports a powerful and flexible routing system. For configuring routing and page layouts, refer to the Next.js documentation. The documentation provides detailed guides to help you define routes and configure page layouts, ensuring your application has an intuitive and powerful user interface. Next.js enables both server-side and client-side components, allowing you to build a seamless user experience.
+
+3. **UI Styling Options**:
+   The Next.js app supports various UI styling solutions including Tailwind CSS, CSS-in-JS (like styled-components or emotion), and traditional CSS. This flexibility allows you to choose the best styling approach that fits your project requirements and developer preferences.
+
+4. **State Management**:
+   For state management, the default solution provided is Zustand. Zustand is a small, fast, and scalable state-management solution that works well with React and Next.js. It simplifies managing global state in your application without the boilerplate often associated with other state management libraries.
+
+5. **Server-side and Client-side Components**:
+   Next.js allows you to utilize both server-side and client-side components. This hybrid approach can optimize performance and user experience by pre-rendering parts of your application on the server while enabling dynamic interactions on the client.
+
+6. **Scaffold-Stark Configuration**:
+   The configuration for Scaffold-Stark is contained within various folders such as `utils`, `web3`, and `components` under the `scaffold-stark` directory. These configurations handle different aspects of the project, including utilities for smart contract interactions, web3 setup, and reusable UI components tailored for Starknet integration.
```

Comment:
Note that these three folders do not contain configurations. They contain helper functions that implement the scaffold-stark framework itself, thus should mostly be untouched unless the users know what they are doing (if the use-case is to create their own dApp). The config file is actually `scaffold.config.ts`: https://github.com/Quantum3-Labs/scaffold-stark-2/blob/main/packages/nextjs/scaffold.config.ts